ipldutil/traverser.go

```diff
@@ -78,6 +83,16 @@ func (tb TraversalBuilder) Start(parentCtx context.Context) Traverser {
 	if tb.Chooser != nil {
 		t.chooser = tb.Chooser
 	}
+	if tb.LinkSystem.DecoderChooser == nil {
+		t.linkSystem.DecoderChooser = defaultLinkSystem.DecoderChooser
+	}
+	if tb.LinkSystem.EncoderChooser == nil {
+		t.linkSystem.EncoderChooser = defaultLinkSystem.EncoderChooser
+	}
+	if tb.LinkSystem.HasherChooser == nil {
+		t.linkSystem.HasherChooser = defaultLinkSystem.HasherChooser
+	}
```

Comment:
I think these might be kind of excessive?  LinkSystem should give reasonable error messages if these are unset, so I think it's reasonable to presume their existence.  Not at all a strongly held opinion though; if this made an API easier to use, carry on.